File: WebShop_NULL/QueryableExtensions.cs
using System.Linq;
using Microsoft.EntityFrameworkCore;
using WebShop_NULL.Models.DomainModels;

namespace WebShop_NULL
{
    public static class QueryableExtensions
    {
        public static IQueryable<User> ById(this IQueryable<User> userSet, int id)
        {
            return userSet.Where(user => user.Id == id);
        }
        
        public static IQueryable<ImageMetadata> ImageById(this IQueryable<User> userSet, int id)
        {
            return userSet.ById(id).Select(user => user.Image);
        } 
    }
}

File: WebShop_FSharp/QueryableExtensions.fs
namespace WebShop_FSharp

open System.Linq
open System.Runtime.CompilerServices
open DomainModels

[<Extension>]
type QueryableExtensions =
    [<Extension>]
    static member inline ById(userSet: IQueryable<User>, id: int) = userSet.Where(fun user -> user.Id = id)
    
    [<Extension>]
    static member inline ImageById(userSet: IQueryable<User>, id: int) = userSet.ById(id).Select(fun user -> user.Image)
    
    [<Extension>]
    static member inline ById(productSet: IQueryable<Product>, id: int) = productSet.Where(fun product -> product.Id = id)
    
    [<Extension>]
    static member inline ImageById(productSet: IQueryable<Product>, id: int) = productSet.ById(id).Select(fun product -> product.Image)
